include/aligner/seed_finder.hpp

```diff
@@ -246,10 +246,11 @@ class seed_finder
 
     // // Populate the seeds given a list of MEMs
     void populate_seeds(
-        std::vector<mem_t> &mems)
+        std::vector<mem_t> &mems,
+        size_t init_mem_size = 0)
```

Comment:
I do not agree with this logic. Populate seeds is meant to populate all mem's in the vector. Passing the starting point changes the meaning of the function.

I would rather change the way it is used in the code. 

Originally the idea was to first collect the list of MEMs (whether filtered or not) and the extract the seeds. At some point this has changed. I would prefer to roll back to the previous behaviour. Otherwise you need to change the semantics of of the functions and their naming, e.g. append_seeds or something like that.